app/services/student_removal_service.rb

```diff
@@ -1,42 +1,63 @@
 # frozen_string_literal: true
 
 class StudentRemovalService
-  def initialize(students:, school:, remove_from_profile: false, token: nil)
-    @students = students
+  class NoSchoolError < StandardError; end
+  class NoClassesError < StandardError; end
+  class StudentHasProjectsError < StandardError; end
+  class NoopError < StandardError; end
+
+  def initialize(school:, remove_from_profile: false, token: nil, raise_on_noop: false)
     @school = school
     @remove_from_profile = remove_from_profile
     @token = token
+    @raise_on_noop = raise_on_noop
   end
 
-  # Returns an array of hashes, one per student, with details of what was removed
-  def remove_students
-    results = []
-    @students.each do |user_id|
-      result = { user_id: }
-      begin
-        # Skip if student has projects
-        projects = Project.where(user_id: user_id)
-        result[:skipped] = true if projects.length.positive?
-
-        unless result[:skipped]
-          ActiveRecord::Base.transaction do
-            # Remove from classes
-            class_assignments = ClassStudent.where(student_id: user_id)
-            class_assignments.destroy_all
-
-            # Remove roles
-            roles = Role.student.where(user_id: user_id)
-            roles.destroy_all
-          end
-
-          # Remove from profile if requested
-          ProfileApiClient.delete_school_student(token: @token, school_id: @school.id, student_id: user_id) if @remove_from_profile && @token.present?
-        end
-      rescue StandardError => e
-        result[:error] = "#{e.class}: #{e.message}"
-      end
-      results << result
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def remove_student(student_id)
+    raise NoSchoolError, 'School not found' if @school.nil?
+    raise NoClassesError, 'School has no classes' if @school.classes.empty?
```

Comment:
Is this something that is important to check?  I don't think the caller cares if there are no classes.  They might want the student removing from the school roles, or from profile.